CMakeLists.txt

```diff
@@ -10,6 +10,28 @@ if(POLICY CMP0076)
     cmake_policy(SET CMP0076 NEW)
 endif()
 
+add_executable(${APP_TARGET})
+
+# Create Mbed OS library
+add_library(mbed-os OBJECT)
+
+include(${MBED_ROOT}/../.mbedbuild/mbed_config.cmake)
+include(${MBED_ROOT}/cmake/toolchain.cmake)
+include(${MBED_ROOT}/cmake/core.cmake)
+include(${MBED_ROOT}/cmake/profile.cmake)
+include(${MBED_ROOT}/cmake/util.cmake)
+
+set_target_properties(mbed-os PROPERTIES MBED_TARGET_LABELS "${MBED_TARGET_LABELS}")
+target_compile_definitions(mbed-os PUBLIC ${MBED_TARGET_DEFINITIONS})
+target_compile_definitions(mbed-os PUBLIC ${MBED_CONFIG_DEFINITIONS})
+
+# Specify a default build type
+if(NOT CMAKE_BUILD_TYPE)
```

Comment:
I'll remove this in #13346

Reply:
I'll wait when your PR is merged so I can update this as well. I am adding a note to point the dependency on your PR.